junit5/src/main/java/org/jboss/weld/junit5/WeldCustomizer.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2018, Red Hat, Inc., and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.weld.junit5;
+
+import org.jboss.weld.environment.se.Weld;
+
+/**
+ * If no {@link WeldInitiator} field annotated with {@link WeldSetup} is present on a test class, all service providers of this interface are used to customize
+ * the "default deployment".
+ *
+ * <p>
+ * A system property with key equal to FQCN of a customizer class may be used to disable a customizer completely. E.g. for a class
+ * {@code org.weld.FooWeldCustomizer} use {@code -Dorg.weld.FooWeldCustomizer=false} to disable the customizer.
```

Comment:
Just thinking out loud - this enables/disables customizer for all your tests. What if you have two "kinds" of tests, one of which should use automagic and the other some of your own customizers?

Reply:
A customizer could implement additional on/off logic. E.g. the annotation-based customizer can skip processing if e.g. the test class is annotated with `@SkipAnnotationCustomizer`.